src/python/dxpy/sugar/context_handlers.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import print_function
+import os
+import sys
+import tempfile
+import subprocess
+import contextlib
+import shutil
+
+
+@contextlib.contextmanager
+def set_env(**environ):
```

Comment:
I suggest two changes:
1. Have this function take a dict arg, rather than **kwargs
2. Add a 'clear' kwarg that controls whether the existing environment is cleared or preserved when updating to the new environment.

Reply:
I think using **kwargs makes it more readable, since you can do:
`with set_env(PYTHONPATH=''): ...` or pass in a dictionary with multiple environment variables to update.

Having a "clear" or "overwrite" option is a good idea.
@Damien-Black Do you have an opinion on this?

Reply:
This is the last sticking point for me - I think it's important to be able to choose whether you want to start from the current environment or have a completely fresh environment. But I'm also fine with finishing this PR and adding that in a separate PR.

Reply:
I think that's an important distinction. I'll add that in this PR.

Reply:
This PR looks good. Once we add a `clear` option to `set_env` we should move to merge it in.